### ISSUE TITLE
performance: chatbox frame-drop when switching to channels with large history

### ISSUE THREAD

**Issue by SilverShardOnline** [AscensionGameDev/Intersect-Engine#1684]
### Description

When a player is moving and clicks the 'All' tab on the chat box, the game screen will freeze from a brief moment.
This does not occur with any other tabs.

Reported on 3 different machines (PC + Mac)

Video of performance issue:
https://youtu.be/cVSh3KnRBzs

### Steps to Reproduce

1. Move the character while clicking over to the 'All' chat tab.

### Version with performance issue

0.7.1.88 

### Last version that worked well

never noticed

### Affected platforms

Windows, macOS

### Did you find any workaround?

No

### Relevant log output

```shell
None.
```

### Duplicate Performance issue Check

- [X] This Performance issue report is not a duplicate to the best of my knowledge.

**Comment by Uyarrr** [comment]
Same issue, it not only happens while switching to All though, it happens when there are text in any channel. I guess, more text history delays longer.

**Comment by Arufonsu** [comment]
Have you guys tried with smaller (in terms of size/quality) fonts? this issue it's honestly present on every single game with large chat history boxes (even some "AAA+" out there), i've happened to fix these by modding clients into using bitmap font for the chat boxes and that usually fix any frame drop when switching tabs.

Anyways, i don't think that this deserves the tag of a "bug" 🤔 This report _doesn't exposes something that is not working_, instead, this is really a performance thing (which could be improved for sure) that's handled differently on each of our own end machines in relation to how much text they can quickly load up when switchings tabs, this is sadly, pretty much relative on each different case scenario, here i haven't personally seen this issue with my project's font, _and i have rarely seen it with the default font only if the chatbox has large amount of text_.

With that said, i've changed the tag to _performance_ for now, not sure if this is the way to go but in this case it feels like it finally suits the report, maybe a good way to go would be to code it so the history that's far too "old" cleans up automatically, this could be configurable or perhaps, we could add some button around the chatbox to wipe it up so we don't have to relog in order to do so 🤔 ... Wll, those are just some ideas anyways. not sure if this should be turned into a discussion at this point @lodicolo 

Regards, and thanks for pointing this out 🙌

**Comment by Arufonsu** [comment]
Not sure if _resolved_ from the root but, the tools from #1798 should help with the needs behind this performance issue, so i'm closing this _for now_ as completed 😸

**Comment by WeylonSantana** [comment]
in fact after #1798 only the action of type enter to enter on "chat mode", freeze the whole client for more or less <100ms
so there is something to look about this

**Comment by Arufonsu** [comment]
> in fact after #1798 only the action of type enter to enter on "chat mode", freeze the whole client for more or less <100ms so there is something to look about this

What i provided may not *solve* the root issue but what it really do is in fact to clean up the additional created gwen elements per line of chat sent within the chat log, and by doing so it's been confirmed to reduce the graphic performance issues / delay when switching between tabs.

What you mention here is not entirely related to this issue report (you are talking about the action of pressing enter to chat, while this issue report was about the chat log and tab switching), plus, i'm unable to reproduce this at all, so could you please show us a video with this behavior that you are reporting **within a new issue report** where you explain this?
